ostelco-ios-client/Extensions/UIViewController+StartApplePay.swift

```diff
@@ -9,25 +9,80 @@
 import Stripe
 import Siesta
 
-extension UIViewController {
-    func startApplePay(product: Product, delegate: PKPaymentAuthorizationViewControllerDelegate) {
-        let merchantIdentifier = Environment().configuration(.AppleMerchantId)
-        // TODO: Consult with Payment Service Provider (Stripe in our case) to determine which country code value to use
-        // https://developer.apple.com/documentation/passkit/pkpaymentrequest/1619246-countrycode
-        let paymentRequest = Stripe.paymentRequest(withMerchantIdentifier: merchantIdentifier, country: product.country, currency: product.currency)
+enum ApplePayError: Swift.Error {
+    // Apple pay support related Errors
+    case unsupportedDevice
+    case noSupportedCards
+    case otherRestrictions
+    // Other errors during payment
+    case userCancelled
+    case primeAPIError(RequestError)
+}
 
-        if !PKPaymentAuthorizationViewController.canMakePayments() {
-            self.showAlert(title: "Payment Error", msg: "Your device does not support apple pay")
-            return
+extension ApplePayError: RawRepresentable {
+    typealias RawValue = String
+
+    // Dont allow to construct using string
+    public init?(rawValue: RawValue) {
+        return nil;
+    }
+
+    public var rawValue: RawValue {
```

Comment:
I would actually recommend just making `ApplePayError` conform to `LocalizedError`, and having these be `localizedDescription` (we can actually localize them later) - `rawValue` should not take on a different meaning here than it does in other places

Reply:
done